frontend.md

```diff
@@ -114,25 +114,185 @@ Object.keys(person).forEach(lang.hitch(this, function(prop){
+### General
+
+We use [compass][compass], which in turn uses [sass][sass]
+
+#### Flexbox for Layout & Flow
+
+Forget about the standard box model; use flexbox. Compass provides [flexbox mixins][flexbox-mixins]. For a visual overview, see [CSS Tricks' flexbox guide][flexbox-guide].
+
+#### Cross-browser support
+
+Use [compass' mixins][compass-mixins] for cross-browser support:
+
+```sass
+#header
+  @include border-bottom-radius(50px)
+```
+
+#### Comments
 
 Comments should on a line of their own (above the target line); not inline.
 
 ```sass
-.country
+.fa-MyComponent
   font-size: 12pt
   // This country has big borders. <-- good comment
   border: 100px
   margin: 0 // but no margins. <-- bad comment
 ```
 
-Use [compass' mixins][compass-mixins] for cross-browser support:
+
+### Component & Widget styling: Naming conventions
+
+This naming convention relies on _structured class names_ and _meaningful hyphens_ (i.e., not
+using hyphens merely to separate words). This helps to work around the current
+limits of applying CSS to the DOM (i.e., the lack of style encapsulation), and
+to better communicate the relationships between classes.
+
+Migration to this architecture will take time and should be done whenever styles of compontents are worked on.
+
+#### Credit
+
+The following styles are heavily based on, and an adaption of the excellent [SUIT CSS naming conventions][suitcss-naming-conventions].
+
+### Components/Widgets
+
+The SASS responsible for component-specific styling.
+
+Syntax: `[<namespace>-]<ComponentName>[--modifierName|-descendentName]`
+
+This has several benefits when reading and writing HTML and CSS/SASS:
+
+* It helps to distinguish between the classes for the root of the component,
+  descendent elements, and modifications.
+* It keeps the specificity of selectors low.
+* It helps to decouple presentation semantics from document semantics.
+
+Your components should each be defined in their own SASS file.
+
+#### Namespace
+
+Components can (and probably should) be prefixed with a namespace. For example, you may wish to avoid the potential for collisions between libraries and your custom components by prefixing all your components with a namespace.
 
 ```sass
-.country
-  @include border-bottom-radius(50px)
+.fa-Button
+  /* … */
+.fa-Tabs
+  /* … */
+```
+
+This makes it clear, when reading the HTML, which components are part of your
+library.
+
+<a name="ComponentName"></a>
+#### ComponentName
+
+The component's name must be written in pascal case. Nothing else in the
+HTML/CSS uses pascal case.
+
+```sass
+.fa-MyComponent
+  /* … */
+```
+
+```html
+<article class="fa-MyComponent">
+  …
+</article>
+```
+
+<a name="ComponentName--modifierName"></a>
+#### ComponentName--modifierName
+
+A component modifier is a class that modifies the presentation of the base
+component in some form (e.g., for a certain configuration of the component).
```

Comment:
I don't think we mean `e.g.` here, it doesn't make sense to me. I think it can be dropped and the above can simply be:
```
(for a certain configuration of the component)
```